### ISSUE TITLE
Switch to IRegistrationContext::registerAlternativeLogin

### ISSUE THREAD

**Issue by julien-nc** [nextcloud/user_oidc#762]
As mentioned in https://github.com/nextcloud/user_oidc/blob/4db5bc0/lib/AppInfo/Application.php#L113
`OC_App::registerLogIn` is deprecated since NC 20.

The new way to register a login method (leading to an extra button in the login form) is to call `IRegistrationContext::registerAlternativeLogin` in the app's `Application::register`.

Problems:
* Getting the db connection (`\OC::$server->get(ProviderMapper::class);` might work)
* We can only register classes with the new system. How are we going to register the same class multiple times but make it point to different oidc providers?

@juliushaertl Not very urgent but at some point the deprecated `OC_App::registerLogIn` will be gone.

**Comment by juliushaertl** [comment]
> We can only register classes with the new system. How are we going to register the same class multiple times but make it point to different oidc providers?

Yeah that one is tricky, I don't see a good way of migrating without adjustments to the server API for that.

**Comment by CarlSchwan** [comment]
This has been done, we now have IAlternativeLoginProvider in server and it's used by OIDC and soon SAML too
